src/v2/normalized_packages.py

```diff
@@ -1,9 +1,30 @@
 """Abstraction for various response models used in V2 implementation."""
-
+import logging
 from collections import defaultdict
 from typing import List, Tuple, Dict, Set
+
+from werkzeug.exceptions import BadRequest
+
 from src.v2.models import Package, Ecosystem
 from f8a_utils.tree_generator import GolangDependencyTreeGenerator
+from f8a_utils.gh_utils import GithubUtils
+
+
+logger = logging.getLogger(__name__)
+gh = GithubUtils()
+
+
+def get_golang_metadata(package) -> Tuple[str, str, str, Dict]:
+    """Clean Package Name, Pkg version & get Golang package_module and version_map."""
+    if "@" not in package.name:
+        raise BadRequest("Package name should be of format package@module.")
+
+    package_name, package_module = package.name.split("@")
+    _, package_version = GolangDependencyTreeGenerator.clean_version(package.version)
+    version_map = {}
+    if gh.is_pseudo_version(package_version):
+        version_map[package_name] = package_version
```

Comment:
Having a version map inside NormalizedPackage violates the SRP rule, This could be moved to `get_batch_sa_data_for_pseudo_version` as a local var.

Reply:
I again have to loop over each Package in `get_batch_sa_data_for_pseudo_version` to generate a version map..?
Are we Okay with Overhead..?

Reply:
Since this is specific to pseudo version impl, moving as much into the specialised code makes sense than altering the generic code.

```
for ..
  op1
for ..
  op2
```
==
```
for ..
  op1
  op2 
```

Complexity of both will be equal and approach#1 would be slightly better because of good [locality of reference](https://en.wikipedia.org/wiki/Locality_of_reference).